mutation/mutation_partition.cc

```diff
@@ -2251,24 +2247,11 @@ to_data_query_result(const reconcilable_result& r, schema_ptr s, const query::pa
     auto consumer = compact_for_query_v2<query_result_builder>(*s, gc_clock::time_point::min(), slice, max_rows,
             max_partitions, query_result_builder(*s, builder));
     auto compaction_state = consumer.get_state();
-    const auto reverse = slice.is_reversed() ? consume_in_reverse::yes : consume_in_reverse::no;
-
-    // FIXME: frozen_mutation::consume supports only forward consumers
-    if (reverse == consume_in_reverse::no) {
-        frozen_mutation_consumer_adaptor adaptor(s, consumer);
-        for (const partition& p : r.partitions()) {
-            const auto res = co_await p.mut().consume_gently(s, adaptor);
-            if (res.stop == stop_iteration::yes) {
-                break;
-            }
-        }
-    } else {
-        for (const partition& p : r.partitions()) {
-            auto m = co_await unfreeze_gently(p.mut(), s);
-            const auto res = co_await std::move(m).consume_gently(consumer, reverse);
-            if (res.stop == stop_iteration::yes) {
-                break;
-            }
+    frozen_mutation_consumer_adaptor adaptor(s, consumer);
+    for (const partition& p : r.partitions()) {
+        const auto res = co_await p.mut().consume_gently(s, adaptor);
+        if (res.stop == stop_iteration::yes) {
+            break;
```

Comment:
I don't understand how you can drop this. If the data came from an old node, it would be a half-reversed mutation, no?

Reply:
Yes it is but same as for the `read_command` I reverse mutations before they are applied. Thus I do not need to distinguished between versions. When applied, it will already be in the native reverse.

Reply:
I don't understand the answer.

Reply:
When the data comes from an old node it is first reversed in `abstract_read_executor::make_mutation_data_request` https://github.com/scylladb/scylladb/blob/b9f7e2f45dd4d20695f1f5f94fe68e42868c80d6/service/storage_proxy.cc#L5039
So I can treat it as it came from a new node.

Reply:
Very good. The patch structure now makes it visible to a diligent reader (though I admit I'm not diligent enough here).